cfn-resources/test/e2e/cluster/cluster_pause_test.go

```diff
@@ -0,0 +1,215 @@
+	})
+
+	t.Run("Unpause Cluster", func(t *testing.T) {
+		testUpdatePauseState(t, testCtx, false)
+	})
+
+	t.Run("Delete Stack", func(t *testing.T) {
+		testDeletePauseStack(t, testCtx)
+	})
+}
+
+func setupPauseSuite(t *testing.T) *pauseTestContext {
+	t.Helper()
+	t.Log("Setting up pause suite")
+	testCtx := new(pauseTestContext)
+	testCtx.setUp(t)
+	return testCtx
+}
+
+func (c *pauseTestContext) setUp(t *testing.T) {
+	t.Helper()
+	c.resourceCtx = utility.InitResourceCtx(pauseStackName, pauseRandSuffix, resourceTypeName, resourceDirectory)
+	c.cfnClient, _ = utility.NewClients(t)
+	_, c.atlasClient20231115014 = utility.NewClients20231115014(t)
+
+	utility.PublishToPrivateRegistry(t, c.resourceCtx)
+	c.setupPrerequisites(t)
+}
+
+func (c *pauseTestContext) setupPrerequisites(t *testing.T) {
+	t.Helper()
+	t.Cleanup(func() {
+		cleanupPausePrerequisites(t, c)
+		cleanupPauseResources(t, c)
+	})
+	t.Log("Setting up prerequisites for pause test")
+
+	var projectID string
+	if projectIDEnvVar := os.Getenv("MONGODB_ATLAS_PROJECT_ID"); projectIDEnvVar != "" {
+		t.Logf("using projectID from env var %s", projectIDEnvVar)
+		projectID = projectIDEnvVar
+	} else {
+		projectID = utility.CreateProject(t, c.atlasClient20231115014, pauseOrgID, pauseProjectName)
+	}
+
+	c.clusterTmplObj = pauseTestCluster{
+		Name:             pauseClusterName,
+		ProjectID:        projectID,
+		Profile:          pauseProfile,
+		Paused:           "false",
+		ResourceTypeName: os.Getenv("RESOURCE_TYPE_NAME_FOR_E2E"),
+	}
+
+	var err error
+	c.template, err = newPauseCFNTemplate(c.clusterTmplObj)
+	utility.FailNowIfError(t, "Error while reading pause CFN Template: %v", err)
+	t.Logf("Pause test setup complete. ProjectID: %s, ClusterName: %s", c.clusterTmplObj.ProjectID, c.clusterTmplObj.Name)
+}
+
+func newPauseCFNTemplate(tmpl pauseTestCluster) (string, error) {
+	return utility.ExecuteGoTemplate(pauseCfnTemplatePath, tmpl)
+}
+
+func testCreatePauseStack(t *testing.T, c *pauseTestContext) {
+	t.Helper()
+	t.Logf("Creating pause stack with template:\n%s", c.template)
+
+	output := utility.CreateStack(t, c.cfnClient, pauseStackName, c.template)
+	clusterID := getPauseClusterIDFromStack(output)
+
+	cluster := readPauseClusterFromAtlas(t, c)
+
+	a := assert.New(t)
+	a.Equal(cluster.GetId(), clusterID)
+	a.False(cluster.GetPaused())
+}
+
+func testUpdatePauseState(t *testing.T, c *pauseTestContext, pause bool) {
+	t.Helper()
+
+	if pause {
+		c.clusterTmplObj.Paused = "true"
+	} else {
+		c.clusterTmplObj.Paused = "false"
+	}
+
+	var err error
+	c.template, err = newPauseCFNTemplate(c.clusterTmplObj)
+	utility.FailNowIfError(t, "Error while reading pause CFN Template: %v", err)
+
+	output := utility.UpdateStack(t, c.cfnClient, pauseStackName, c.template)
+	_ = getPauseClusterIDFromStack(output)
+
+	cluster := readPauseClusterFromAtlas(t, c)
+
+	a := assert.New(t)
+	if pause {
+		a.True(cluster.GetPaused())
+	} else {
+		a.False(cluster.GetPaused())
+	}
```

Comment:
```suggestion
	assert.Equal(t, pause, cluster.GetPaused())
```

nit: i think this is a bit clearer and concise